### PR TITLE
chore: consolidate MCP tool-name parsing into internal/mcpname

### PR DESCRIPTION
**Pull request by vishalg0wda**
## What

Commit `2e738a79a` (AGE-2749) introduced the `mcp__<server>__<function>` / `MCP:` tool-name parsers, but they lived in the heavy `risk_analysis` Temporal-activity package while the **same `strings.SplitN(name, "__", 3)` logic was hand-rolled in 5 more places** in `hooks/` (the shadow-MCP guard + per-host telemetry attribution) — with subtle drift in empty-part handling.

This extracts the four parsers — `IsMCPToolName`, `MCPFunctionOf`, `MCPServerOf`, `AttributeTool` — into a new **dependency-free `internal/mcpname` package** (imports only `strings`) and routes every caller through it:

- **risk_analysis**: `llm_judge.go`, `analyze_batch.go`
- **hooks**: `mcp_match.go` (`parseClaudeToolName`), `shadow_mcp_access.go` (`mcpServerIdentityFromToolName`), `codex_hooks.go`, `cursor_hooks.go`, `pending_helpers.go`

The old `risk_analysis/tool_names.go` and its internal test are removed; their coverage now lives in `internal/mcpname/mcpname_test.go`.

## Notable details

- **Cursor `MCP:` collision** — `AttributeTool` recognizes **both** the Claude-style `mcp__` form and Cursor's `MCP:` prefix. `cursor_hooks.go` has dedicated downstream `MCP:` handling, so its telemetry block keeps an explicit `mcp__` guard; likewise `hooks.parseClaudeToolName` guards on the `mcp__` prefix so Cursor names aren't parsed as Claude entries.
- **Validation hardening** — a well-formed name now requires both segments: `mcp__<server>__<function>` needs a non-empty server *and* function, and `MCP:` needs a non-empty suffix. Malformed names (`mcp____x`, a bare `MCP:`) are no longer classified as MCP-routed, so they can't leak into MCP-only attribution/enforcement paths.

## Verification

- `go build ./...` ✅
- `internal/mcpname` + `hooks` MCP/shadow tests + `risk_analysis` judge tests ✅ (incl. new edge-case coverage)
- `mise lint:server` ✅
- No `SplitN(…, "__", 3)` copies remain outside `mcpname`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by vishalg0wda** [comment]
Thanks @cubic-dev-ai — addressed in 33c72cc68:

- **P2 (empty `MCP:`)** — `IsMCPToolName` now requires a non-empty suffix after `MCP:`.
- **P2 (empty `mcp__` server)** — now requires both `<server>` and `<function>` segments non-empty. Also restored an explicit `mcp__` prefix guard in `hooks.parseClaudeToolName` so Cursor `MCP:` names (which the shared parser also recognizes) aren't parsed as Claude entries. Added edge-case tests in both `mcpname_test.go` and `mcp_match_test.go`.
- **P3 (delete empty placeholder)** — `tool_names.go` and `mcp_match_internal_test.go` are gutted because their contents moved to `internal/mcpname`. This environment's security policy blocks me from running `rm`/`git rm`, so the deletion needs a maintainer; their coverage already lives in `mcpname_test.go`.

File: server/internal/background/activities/risk_analysis/tool_names.go [hunk 1]
```diff
@@ -1,63 +1,6 @@
...
-func AttributeTool(name string) (server, function string, isMCP bool) {
-	if !IsMCPToolName(name) {
-		return "", "", false
-	}
-	return MCPServerOf(name), MCPFunctionOf(name), true
-}
+// The MCP tool-name parsers that lived here (IsMCPToolName, MCPFunctionOf,
+// MCPServerOf, AttributeTool) moved to the internal/mcpname package so the
+// hooks package can share them without depending on this Temporal-activity
+// package. This file is intentionally empty and should be deleted.
```
[the remark this answers is not in the document]
Comment (vishalg0wda) [reply server/internal/background/activities/risk_analysis/tool_names.go]: Valid nit. `tool_names.go` and `mcp_match_internal_test.go` are empty because their contents moved to `internal/mcpname` (coverage now in `mcpname_test.go`). The agent that made this change is hard-blocked from `rm`/`git rm` by this environment's "Production Destructive Actions" policy, so the deletion needs a maintainer:

```
git rm server/internal/background/activities/risk_analysis/tool_names.go server/internal/background/activities/risk_analysis/mcp_match_internal_test.go
```

Leaving this thread open as the maintainer follow-up.

**Review by subomi — APPROVED** [review]
